src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -152,26 +177,52 @@ const HeatmapPlot = (props) => {
     setMaxCells(Math.floor(width * 0.8));
   }, [width]);
 
-  const handleMouseOver = (...args) => {
-    if (args.length < 2) {
-      return;
-    }
-
-    if ('x' in args[1] && 'y' in args[1]) {
-      hoverCoordinates.current = {
-        x: args[1].x,
-        y: args[1].y,
-      };
-    }
-
-    if (!args[1].datum) {
-      return;
-    }
+  useEffect(() => {
+    dispatch(updateCellInfo({ cellName: cellHighlight }));
+  }, [cellHighlight]);
+
+  const buildExpressionMatrix = () => {
+    const cellIds = heatmapData.cellOrder.map((x) => `${x}`);
+    const genes = heatmapData.geneOrder;
+
+    // array with shape [gene_1 cell_1, ..., gene_1 cell_n, gene_2 cell_1, ... ]
+    const geneOrderedExpression = heatmapData.heatmapData.map((x) => x.expression);
+
+    // first convert to cell by gene matrix
+    const cellByGeneMatrix = listToMatrix(geneOrderedExpression, cellIds.length);
+
+    // scale so that each gene has minimum 0 max 255
+    const scaledCellByGeneMatrix = cellByGeneMatrix.map((row) => {
+      const geneMin = Math.min(...row);
+      const geneMax = Math.max(...row);
+
+      return row.map((x) => convertRange(x, [geneMin, geneMax], [0, 255]));
+    });
+
+    // vitesse Heatmap uses:
+    // array with shape [cell_1 gene_1, ..., cell_1 gene_n, cell_2 gene_1, ... ]
```

Comment:
gene_x numbers are messed up in this comment

Reply:
why? it is all genes for first cell, then all genes for second cell, etc.

Reply:
no, the comment is right, I got confused